cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -125,35 +128,43 @@ class CompareSorter {
           std::stable_partition(indices_begin, indices_end,
                                 [&values](uint64_t ind) { return !values.IsNull(ind); });
     }
-    std::stable_sort(indices_begin, nulls_begin,
-                     [&values](uint64_t left, uint64_t right) {
-                       return values.GetView(left) < values.GetView(right);
-                     });
+    if (options.order == SortOrder::ASCENDING) {
+      std::stable_sort(indices_begin, nulls_begin,
+                       [&values](uint64_t left, uint64_t right) {
+                         return values.GetView(left) < values.GetView(right);
+                       });
+    } else {
+      std::stable_sort(indices_begin, nulls_begin,
+                       [&values](uint64_t left, uint64_t right) {
+                         return values.GetView(left) > values.GetView(right);
```

Comment:
This requires that all of `==`, `<`, and `>` are implemented in the value type.
Instead, writing `right < left` here only requires `==` and `<`.
IMHO, requiring only `==` and `<` is better to support extension types in the future.

Reply:
Good catch!